keras_nlp/src/models/stable_diffusion_3/mmdit_block.py

```diff
@@ -55,7 +55,7 @@ def __init__(
             epsilon=1e-6,
             center=False,
             scale=False,
-            dtype=self.dtype_policy,
+            dtype="float32",
```

Comment:
For layernorm we already have some code to avoid ever computing our mean/variance at half precision. https://github.com/keras-team/keras/blob/e7b5a5d6ebac694aa629ef20e104671f40999f2f/keras/src/layers/normalization/layer_normalization.py#L203-L206

Is this needed?

Reply:
At first glance, it seemed reasonable to set `self.dtype_policy` but in fact, it led to overflow/underflow.

The root cause is that `LayerNormalization` computes with float16 when `self.dtype` is float16:
https://github.com/keras-team/keras/blob/e7b5a5d6ebac694aa629ef20e104671f40999f2f/keras/src/layers/normalization/layer_normalization.py#L203

I doubt we should fix it upstream. It doesn't makes sense to compute with float16 since it causes numerical instability.